tools/opensnoop.py

```diff
@@ -352,20 +367,37 @@
     if (data.name[0] != '/') { // relative path
         struct task_struct *task;
         struct dentry *dentry;
+        struct vfsmount *vfsmnt;
+        struct mount *mnt, *mnt_parent;
         int i;
 
-        task = (struct task_struct *)bpf_get_current_task_btf();
+        task = (struct task_struct *)bpf_get_current_task();
```

Comment:
We may have some verification with newer kernel. Which kernel did you test?

Reply:
I tested on kernel 5.4/5.14/5.15. When I switched to bpf_get_current_task_btf(), it failed with such error messages:

```
84: (85) call bpf_get_current_task_btf#158
85: (b7) r6 = 31
; dentry = task->fs->pwd.dentry;
86: (79) r9 = *(u64 *)(r0 +2768)
; vfsmnt = task->fs->pwd.mnt;
87: (79) r8 = *(u64 *)(r9 +40)
; dentry = task->fs->pwd.dentry;
88: (07) r9 += 48
; mnt_parent = mnt->mnt_parent;
89: (79) r7 = *(u64 *)(r8 -16)
R8 is ptr_vfsmount invalid negative access: off=-16
processed 87 insns (limit 1000000) max_states_per_insn 0 total_states 4 peak_states 4 mark_read 3
```

Reply:
btw, other tools still use bpf_get_current_task(), and bpf_get_current_task_btf() looks not stable, I didn't analyze it yet.